salt/_states/metalk8s_cordon.py

```diff
@@ -70,4 +69,19 @@ def node_uncordoned(name, **kwargs):
     name
         the name of the node
     '''
+    all_node_taints = __salt__['metalk8s_kubernetes.node_taints'](
+        name, **kwargs
+    )
+    key_to_remove = "node.kubernetes.io/unschedulable"
+
+    new_taints = [
+        t for t in all_node_taints
+        if t['key'] not in key_to_remove
+    ]
+
+    __salt__['metalk8s_kubernetes.node_set_taints'](
```

Comment:
Add an `if` here to not call this one if not needed `if new_taints == all_node_taints`